samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/MainActivity.kt

```diff
@@ -27,19 +28,40 @@ open class MainActivity : AppCompatActivity() {
         const val CLIENT_ID = "12cc4070a481bc73"
         const val REDIRECT_URL = "fxaclient://android.redirect"
         const val CONFIG_URL = "https://latest.dev.lcip.org"
+        const val FXA_STATE_PREFS_KEY = "fxaAppState"
+        const val FXA_STATE_KEY = "fxaState"
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        Config.custom(CONFIG_URL).whenComplete { value: Config ->
-            account = FirefoxAccount(value, CLIENT_ID, REDIRECT_URL)
+        getSharedPreferences(FXA_STATE_PREFS_KEY, Context.MODE_PRIVATE).getString(FXA_STATE_KEY, "").let {
+            FirefoxAccount.fromJSONString(it).then({
+                this.account = it
+                account?.getProfile()
+            }, {
+                Config.custom(CONFIG_URL).whenComplete { value: Config ->
+                    account = FirefoxAccount(value, CLIENT_ID, REDIRECT_URL)
+                }
+                FxaResult()
+            }).whenComplete {
+                val txtView: TextView = findViewById(R.id.txtView)
+                runOnUiThread {
+                    txtView.text = getString(R.string.signed_in, "${it.displayName ?: ""} ${it.email}")
+                }
+            }
         }
 
         findViewById<View>(R.id.button).setOnClickListener {
             account?.beginOAuthFlow(scopes, false)?.whenComplete { openTab(it) }
         }
+
+        findViewById<View>(R.id.buttonPurge).setOnClickListener {
+            getSharedPreferences(FXA_STATE_PREFS_KEY, Context.MODE_PRIVATE).edit().putString(FXA_STATE_KEY, "").apply()
+            val txtView: TextView = findViewById(R.id.txtView)
+            txtView.text = "State cleared!"
```

Comment:
We need to put "State cleared!" into strings.xml

Reply:
Updated to "Logged out!" to be consistent with the button